### PR TITLE
use headless for tests and linting

### PR DESCRIPTION
**Pull request by superloach**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -30,8 +30,10 @@ jobs:
     #   run: go build -v ./...
 
     - name: Test
-      run: go test -v ./...
+      run: go test -v -flags headless ./...
```
Comment (zacharyburkett) [inline .github/workflows/go.yml:33]: I think this needs to be `--tags=headless`, right?

**Review by zacharyburkett — CHANGES_REQUESTED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (superloach) [reply .github/workflows/go.yml:33]: yeah it should, sorry i mixed that up

**Comment by superloach** [comment]
Not sure why it's still pulling glfw with the headless tag. I'll look into this.

**Comment by superloach** [comment]
Oh duh, it's trying to run tests on `internal/ebiten`. We need to add tags to the backends themselves as well.

**Comment by superloach** [comment]
why must this be a pain lol

**Comment by superloach** [comment]
might just.. manually install and run golangci-lint instead of using the action. this is dumb.

**Comment by superloach** [comment]
finally :tada:

**Review by zacharyburkett — APPROVED** [review]

**Pull request merged**